components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -143,6 +153,28 @@ class AtmosphereInput
  *  also outward facing but do not need those respective sections of the AD, 
  *  i.e. during ad.init, ad.run and ad.finalize.
  */
+/* ---------------------------------------------------------- */
+inline void AtmosphereInput::pull_input(const std::string filename, const std::string var_name, const std::vector<std::string> var_dims,
```

Comment:
Rather than forcing the user to figure out the padding, to pass it to the fcn, we could make this fcn templated on the ValueType, and have the last arg be `ValueType* data`. Then, the padding would be computed with
```
constexpr int pack_size = sizeof(ValueType) / sizeof(Real);
const int padding = PackInfo<pack_size>::padding(dim_lens.back());
...
auto data_real = reinterpret_cast<Real*>(data);
grid_read_data_array(filename,var_name,dim_lens,data_length,padding,data_real);
```
The `PackInfo` helper class is in `ekat_pack_utils.hpp`.

Reply:
Btw, I'm assuming that `dim_lens` are the 'physical' dimensions, without the padding, right? If not, then scratch my comment above.